### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by DanWebb**
This PR adds TypeScript typings, based off the [style guide](https://github.com/sindresorhus/typescript-definition-style-guide) and this repos [README](https://github.com/imagemin/imagemin-pngquant/blob/master/readme.md) ❤️

One difference is that `quality` is typed as a `number` or `string` where it's just `string` in the readme. I saw use of `number` in the `test.js` file so thought this must be a documentation error? I'll send another PR to update the README if so.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Check out https://github.com/imagemin/imagemin-pngquant/commits/master I have fixed the types and changed some options. Can you fix the merge conflict and update?

**Comment by DanWebb** [comment]
Thanks Sindre! I'll take a look and update later today.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  strip?: boolean;
+
+  /**
+   * Instructs pngquant to use the least amount of colors required to meet or exceed the max quality. If conversion results in quality below the min quality the image won't be saved.
+   * Min and max are numbers in range 0 (worst) to 1 (perfect), similar to JPEG.
+   * 
+   * @example [0.3, 0.5]
+   * 
+   */
+  quality?: number[];
```
Comment (sindresorhus) [inline index.d.ts]: This should be a tuple.
```suggestion
  quality?: [number, number];
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+export interface Options {
+  /**
+   * Speed `10` has 5% lower quality, but is about 8 times faster than the default. Speed `11` disables dithering and lowers compression level.
```
Comment (sindresorhus) [inline index.d.ts]: This is missing the `1 (brute-force) to 11 (fastest)` part of the docs.

Comment (DanWebb) [reply index.d.ts]: I couldn't see in the style guide how you'd like this added? Just `Values: 1 (brute-force) to 11 (fastest)` on a line underneath or maybe in a `@remarks`?

Comment (sindresorhus) [reply index.d.ts]: I think it's fine to just put it below here on a new line, without `@remarks`.

**Comment by sindresorhus** [comment]
Looks great now. Thanks for working so hard on it :)

**Pull request merged**